### ISSUE TITLE
Document our build artifacts & process

### ISSUE THREAD

**Issue by linsun** [kgateway-dev/community#27]
This would be must to have and I believe required for cncf graduated projects.

- [ ] Document the project's release process.

**Comment by linsun** [comment]
Hi @jenshu is this something you can check if we have already?  The goal is to make sure we have a process for how to build and release if someone else wants to.

**Comment by jenshu** [comment]
> Hi [@jenshu](https://github.com/jenshu) is this something you can check if we have already? The goal is to make sure we have a process for how to build and release if someone else wants to.

@linsun I don't think we have anything formal written up. Can this just be a README in the kgateway repo, or should we add a page somewhere on the docs site (https://kgateway.dev/docs/)?

**Comment by sam-heilbron** [comment]
Completed by: https://github.com/kgateway-dev/kgateway/pull/12205
